timemachine/fe/mcgregor.py

```diff
@@ -362,7 +416,8 @@ def recursion(
     max_cores,
     threshold,
     enforce_core_core,
-    connected_core,
+    max_connected_components: Optional[int],
```

Comment:
This is a tough parameter for me to figure out:

if `max_connected_components=0` or `None` - then we don't actually mean zero connected components, but effectively we are saying `max_connected_components=max(len(g1.nodes), len(g2.nodes))`. I'm wondering if this should default to `np.inf` to be more explicit? 

This would have the benefit of simplifying the short-circuit logic above:

```
if max_connected_components and n_ccs_with_mapped_nodes > max_connected_components:
    return True
```

Reply:
Note that `max_connected_components=0` is not allowed and will raise an `AssertionError`. My reasoning for `None` instead of `np.inf` to indicate no limit on the maximum number of connected components is just that there's no `np.inf` for integers, and `Optional[int]` seemed clearer to me than `int | float`.

Reply:
`np.iinfo(im.dtype).max`?

Reply:
Clarification Q: is the suggestion to remove `None` as an allowed value for the option `max_connected_components`, and instead pass a large integer when we don't want the filter to apply? (Note that there is some short-circuiting logic to skip the check entirely in the case when the filter is a no-op: https://github.com/proteneer/timemachine/blob/3cb2a8d63f8b4b817119a5872790a9c819ee8f14/timemachine/fe/mcgregor.py#L435; this might get a bit messier if we don't allow `None`)

Reply:
Ugh hm good point - I'm not sure the best way to handle this. Anyways I think it's probably fine as-is, some of these comments were done in a drive-by fashion to help me understand what's going on.

Reply:
Will opt to keep as-is since it seems cleaner than the alternatives I've been able to think of.